### PR TITLE
Final Exercise for Function development in a package environment

### PR DESCRIPTION
**Pull request by cwickham**
Very rough, but basic idea is to add two functions to the package: one based on a snippet of code we provide, and one that is only described through functionality and learner needs to figure out code.

Both will need some basic docs, and probably figuring some dependency issues (e.g. snippet might use a function from another package).

Would be really nice if second function requires calling first function to demonstrate building up more complicated functionality, but will depend on final project.

My only concern is, is two functions too much at this stage?  Should we limit it to one...then maybe have some optional extensions.

### CODE REVIEW

File: chapters/04-function-development.Rmd [hunk 1]
```diff
@@ -13,10 +13,36 @@
...
 <!-- NOTE: Preferably something linked to working towards completing the project assignments, maybe to be used as a milestone? -->
 
+1. Add a `FIXME()` function to your FIXME package
+    
+    The following code snippet produces ... FIXME.  
+    ```{r, eval = FALSE}
+    # A line or two of R code that does something useful
+    FIXME
+    ```
+    Turn it into a function called `FIXME()` that takes argument for       `FIXME` and `FIXME` and include it in your FIXME package.
```
Comment (mbonsma) [inline chapters/04-function-development.Rmd:27]: I like this a lot - providing a regular code snippet to turn into a function. This mimics the natural process of code development.

Comment (mbonsma) [reply chapters/04-function-development.Rmd:27]: Do you want to add instructions to create function documentation explicitly? It's implied at the end of question 2, just wondering if it should be part of question 1 as well.

**Review by mbonsma — APPROVED** [review]
This looks great and is about as detailed as it can get right now. I don't think two functions is too much - this is the final chapter assignment after all, and I like that each question works different muscles. 

Just one thought - this is written assuming a pretty rigid correct answer, so if an instructor was to modify the individual project assignment, these questions would need to be modified too. I think that's fine, just something to keep in mind.

File: chapters/04-function-development.Rmd [hunk 2]
```diff
@@ -13,10 +13,36 @@
 - Function documentation (with roxygen2), part 1
 - To use `@examples` to help with creating function
 
-
-
 ## Final exercise
 
 <!-- NOTE: Preferably something linked to working towards completing the project assignments, maybe to be used as a milestone? -->
 
+1. Add a `FIXME()` function to your FIXME package
```
Comment (lwjohnst86) [inline chapters/04-function-development.Rmd:20]: We may need to make this a bit more generic given learners will have different assignment packages. For now its a good starting point.

File: chapters/04-function-development.Rmd (hunk 1 above)
Comment (cwickham) [reply chapters/04-function-development.Rmd]: Good point, I've added an explicit step.

**Pull request merged**